### ISSUE TITLE
False positive for Lint/RedundantSafeNavigation with safe nav on class variable

### ISSUE THREAD

**Issue by dewey-the-composer** [rubocop/rubocop#14279]
When using a safe nav on a class variable (i.e. `self::VARIABLE&.constantize`), rubocop claims "Lint/RedundantSafeNavigation: Redundant safe navigation detected, use . instead."

--------

## Expected behavior

Because it's possible for `self::VARIABLE` to be nil, it should not consider this redundant.

## Actual behavior

Rubocop claims that the safe nav on the code was unnecessary. Please see the code in "Steps to reproduce the problem"

## Steps to reproduce the problem

Code that can reproduce the issue:
```
class MyClass
  VARIABLE = nil
  class << self
    def constantize_variable
      self::VARIABLE&.constantize
    end
  end
end
```
Run:
```
bundle exec rubocop -a --force-exclusion
```
Receive:
```
Offenses:

app/models/my_class.rb:5:21: W: [Correctable] Lint/RedundantSafeNavigation: Redundant safe navigation detected, use . instead.
    self::VARIABLE&.constantize
```
In contrast, the following does NOT throw the error:
```
class MyClass
  def test
    variable = nil
    variable&.constantize
  end
end
```
--
Why am I doing this code? I have children of `MyClass` that override `self::VARIABLE` with their own settings (I don't remember why I decided to do settitngs this way, probably something to do with STI), so the parent defaults to `nil` but I constantize whatever the child provides to set a class for VARIABLE.

## RuboCop version

```
bundle exec rubocop -V
1.66.1 (using Parser 3.3.6.0, rubocop-ast 1.32.3, running on ruby 3.3.3) [arm64-darwin23]
  - rubocop-capybara 2.21.0
  - rubocop-factory_bot 2.26.1
  - rubocop-minitest 0.36.0
  - rubocop-performance 1.22.1
  - rubocop-rails 2.26.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.1.0
  - rubocop-rspec_rails 2.30.0

```

**Comment by koic** [comment]
It doesn’t reproduce with the latest RuboCop 1.76. Please upgrade to the latest version.

**Comment by dewey-the-composer** [comment]
> It doesn’t reproduce with the latest RuboCop 1.76. Please upgrade to the latest version.

Thank you! I'll add this upgrade to our backlog
